### PR TITLE
chore: upgrade globals v15

### PR DESCRIPTION
**Pull request by aladdin-add**

### CODE REVIEW

File: tests/eslint-rule-tester.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const defaultConfig = {
     languageOptions: {
         ecmaVersion: 6,
         sourceType: "commonjs",
-        globals: globals.node,
+        globals: { ...globals.es2015, ...globals.node },
```
Comment (scagood) [inline tests/eslint-rule-tester.js:26]: It may be worth using `es2021` here :thinking:

**Comment by aladdin-add** [comment]
Let's wait and see if it can be fixed in eslint: https://github.com/eslint/eslint/discussions/18317

**Comment by aladdin-add** [comment]
it will be fixed in next eslint v9 release: https://github.com/eslint/eslint/commit/0db676f9c64d2622ada86b653136d2bda4f0eee0.

but won't be backported to v8. so I think it's fine to merge it now. :)

File: tests/eslint-rule-tester.js (hunk 1 above)
Comment (aladdin-add) [reply tests/eslint-rule-tester.js:28]: better to be same as the `ecmaVersion`. :)

**Pull request merged**